File: src/main/java/hilfling/backend/hilfling/model/Article.java
package hilfling.backend.hilfling.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity(name = "articles")
@Table(name = "articles")
public class Article implements Serializable, BaseModel {
    public Article() {
    }

    public Article(String title, String content, SecurityLevel securityLevel, PhotoGangBanger photoGangBanger) {
        this.title = title;
        this.content = content;
        this.securityLevel = securityLevel;
        this.photoGangBanger = photoGangBanger;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "content", columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(name = "date_published", insertable = false, updatable = false)
    @ColumnDefault("CURRENT_TIMESTAMP")
    private Date datePublished;

    @ManyToOne
    @JoinColumn(name = "security_level_id", referencedColumnName = "id", nullable = false)
    private SecurityLevel securityLevel;

    @ManyToOne
    @JoinColumn(name = "photo_gang_banger_id", referencedColumnName = "id", nullable = false)
    private PhotoGangBanger photoGangBanger;

}

File: src/main/kotlin/hilfling/backend/hilfling/model/Article.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface Article : BaseModel<Article> {
    companion object : Entity.Factory<Article>()

    var title: String
    var content: String

    // Foreign keys
    var securityLevel: SecurityLevel
    var photoGangBanger: PhotoGangBanger
}

object Articles : BaseTable<Article>("article") {
    val title = varchar("name").bindTo { it.title }
    val content = varchar("content").bindTo { it.content }

    // Foreign keys
    val securityLevel = int("security_level_id").references(SecurityLevels){it.securityLevel}
    val photoGangBanger = int("photo_gang_banger_id").references(PhotoGangBangers){it.photoGangBanger}
}

val Database.articles get() = this.sequenceOf(Articles)